gui/mozregui/build_runner.py

```diff
@@ -74,7 +74,9 @@ def evaluate(self, build_info, allow_back=False):
         else:
             self.evaluate_started.emit('')
 
-    def finish(self, verdict):
+    def finish(self, verdict, options):
+        if options['profile'] and options['profile_persistence'] == 'clone-first':
+            options['profile'].cleanup()
```

Comment:
I think this should be not be called in the launcher.finish method, because this is called after every tested build. I believe you have to put that code in Bisector.stop instead. Also, does "if options['profile']" works in case the profile field is empty ? (Just a question)